File: Source_Code/icw_core_biz/pkg/dto/project/assets.go
package project

type ProjectGroup struct {
	Id        uint64
	Name      string
	SortOrder string
	Images    []*ProjectImage
}

type ProjectImage struct {
	Uuid         string `json:"uuid"`
	FileName     string `json:"file_name"`
	ContentType  string `json:"content_type"`
	SizeBytes    uint64 `json:"size_bytes"`
	Width        uint32 `json:"width"`
	Height       uint32 `json:"height"`
	Metadata     string `json:"metadata"`
	Status       string `json:"status"`
	ThumbnailURL string `json:"thumbnail_url"`
	UploadedAt   string `json:"uploaded_at"`
	CreatedAt    string `json:"created_at"`
}

type ReportProjectImageItem struct {
	ImageUuid string
	Status    string
}

type UploadProjectImageItem struct {
	FileName    string
	ContentType string
	SizeBytes   uint64
	Width       uint32
	Height      uint32
	Metadata    string
}

type UploadProjectImageResult struct {
	Image              *ProjectImage
	OriginalUploadURL  string
	ThumbnailUploadURL string
}

type GetProjectAssetsRequest struct {
	UserId    uint64
	ProjectId uint64
}

type GetProjectAssetsResponse struct {
	Groups []*ProjectGroup
}

type CreateProjectGroupRequest struct {
	UserId    uint64
	ProjectId uint64
}

type CreateProjectGroupResponse struct {
	Group *ProjectGroup
}

type DeleteProjectGroupRequest struct {
	UserId    uint64
	ProjectId uint64
	GroupId   uint64
}

type DeleteProjectGroupResponse struct{}

type MoveProjectGroupRequest struct {
	UserId          uint64
	ProjectId       uint64
	GroupId         uint64
	PreviousGroupId uint64
	NextGroupId     uint64
	MoveToFirst     bool
	MoveToLast      bool
}

type MoveProjectGroupResponse struct {
	Group *ProjectGroup
}

type UpdateProjectGroupRequest struct {
	UserId    uint64
	ProjectId uint64
	GroupId   uint64
	Name      string
}

type UpdateProjectGroupResponse struct {
	Group *ProjectGroup
}

type DeleteProjectImageRequest struct {
	UserId     uint64
	ProjectId  uint64
	ImageUuids []string
}

type DeleteProjectImageResponse struct{}

type GetProjectImageOriginalRequest struct {
	UserId    uint64
	ProjectId uint64
	ImageUuid string
}

type GetProjectImageOriginalResponse struct {
	OriginalURL string
}

type MoveProjectImageRequest struct {
	UserId        uint64
	ProjectId     uint64
	ImageUuids    []string
	TargetGroupId uint64
}

type MoveProjectImageResponse struct {
	Images []*ProjectImage
}

type ReportProjectImageRequest struct {
	UserId    uint64
	ProjectId uint64
	Images    []*ReportProjectImageItem
}

type ReportProjectImageResponse struct {
	Images []*ProjectImage
}

type UploadProjectImageRequest struct {
	UserId    uint64
	ProjectId uint64
	GroupId   uint64
	Images    []*UploadProjectImageItem
}

type UploadProjectImageResponse struct {
	Images []*UploadProjectImageResult
}

File: Source_Code/icw_core_web/src/types/project/assets.ts
export const PROJECT_IMAGE_STATUS_PENDING = 'pending';
export const PROJECT_IMAGE_STATUS_UPLOADED = 'uploaded';
export const PROJECT_IMAGE_STATUS_FAILED = 'failed';

export type ProjectImageStatus =
  | typeof PROJECT_IMAGE_STATUS_PENDING
  | typeof PROJECT_IMAGE_STATUS_UPLOADED
  | typeof PROJECT_IMAGE_STATUS_FAILED;

export interface ProjectGroup {
  id: string;
  name: string;
  sort_order: string;
  images: ProjectImage[];
}

export interface ProjectImage {
  uuid: string;
  file_name: string;
  content_type: string;
  size_bytes: number;
  width: number;
  height: number;
  metadata: string;
  status: ProjectImageStatus;
  thumbnail_url: string;
  uploaded_at: string;
  created_at: string;
}

export interface UploadProjectImageItem {
  file_name: string;
  content_type: string;
  size_bytes: number;
  width: number;
  height: number;
  metadata: string;
}

export interface UploadProjectImageResult {
  image: ProjectImage;
  original_upload_url: string;
  thumbnail_upload_url: string;
}

export interface GetProjectAssetsResponse {
  groups: ProjectGroup[];
}

export interface CreateProjectGroupResponse {
  group: ProjectGroup;
}

export interface DeleteProjectGroupRequest {
  project_id: string;
  group_id: string;
}

export type DeleteProjectGroupResponse = Record<string, never>;

export interface MoveProjectGroupRequest {
  project_id: string;
  group_id: string;
  previous_group_id: string;
  next_group_id: string;
  move_to_first: boolean;
  move_to_last: boolean;
}

export interface MoveProjectGroupResponse {
  group: ProjectGroup;
}

export interface UpdateProjectGroupRequest {
  project_id: string;
  group_id: string;
  name: string;
}

export interface UpdateProjectGroupResponse {
  group: ProjectGroup;
}

export interface DeleteProjectImageRequest {
  project_id: string;
  image_uuids: string[];
}

export type DeleteProjectImageResponse = Record<string, never>;

export interface GetProjectImageOriginalResponse {
  original_url: string;
}

export interface MoveProjectImageRequest {
  project_id: string;
  image_uuids: string[];
  target_group_id: string;
}

export interface MoveProjectImageResponse {
  images: ProjectImage[];
}

export interface ReportProjectImageRequest {
  project_id: string;
  image_uuid: string;
  status: ProjectImageStatus;
}

export interface ReportProjectImageResponse {
  image: ProjectImage;
}

export interface UploadProjectImageRequest {
  project_id: string;
  group_id: string;
  images: UploadProjectImageItem[];
}

export interface UploadProjectImageResponse {
  images: UploadProjectImageResult[];
}
